core/src/main/java/org/apache/accumulo/core/util/ThreadPools.java

```diff
@@ -0,0 +1,368 @@
+        TimeUnit unit, BlockingQueue<Runnable> workQueue, ThreadFactory threadFactory) {
+      super(corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue, threadFactory);
+    }
+
+    @Override
+    public void execute(Runnable command) {
+      super.execute(new TraceRunnable(command));
+    }
+
+    @Override
+    public <T> Future<T> submit(Callable<T> task) {
+      return super.submit(new TraceCallable<T>(task));
+    }
+
+    @Override
+    public <T> Future<T> submit(Runnable task, T result) {
+      return super.submit(new TraceRunnable(task), result);
+    }
+
+    @Override
+    public Future<?> submit(Runnable task) {
+      return super.submit(new TraceRunnable(task));
+    }
+
+    private <T> Collection<? extends Callable<T>>
+        wrapCollection(Collection<? extends Callable<T>> tasks) {
+      List<Callable<T>> result = new ArrayList<Callable<T>>(tasks.size());
+      tasks.forEach(t -> result.add(new TraceCallable<T>(t)));
+      return result;
+    }
+
+    @Override
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> tasks)
+        throws InterruptedException {
+      return super.invokeAll(wrapCollection(tasks));
+    }
+
+    @Override
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> tasks, long timeout,
+        TimeUnit unit) throws InterruptedException {
+      return super.invokeAll(wrapCollection(tasks), timeout, unit);
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks)
+        throws InterruptedException, ExecutionException {
+      return super.invokeAny(wrapCollection(tasks));
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks, long timeout, TimeUnit unit)
+        throws InterruptedException, ExecutionException, TimeoutException {
+      return super.invokeAny(wrapCollection(tasks), timeout, unit);
+    }
+
+    @Override
+    public void shutdown() {
+      super.shutdown();
+    }
+
+    @Override
+    public List<Runnable> shutdownNow() {
+      return super.shutdownNow();
+    }
+
+  }
+
+  public static class CloseableThreadPoolExecutor implements AutoCloseable {
+
+    private final ThreadPoolExecutor tpe;
+
+    public CloseableThreadPoolExecutor(ThreadPoolExecutor tpe) {
+      this.tpe = tpe;
+    }
+
+    @Override
+    public void close() throws Exception {
+      this.tpe.shutdownNow();
+    }
+
+  }
+
+  // the number of seconds before we allow a thread to terminate with non-use.
+  public static final long DEFAULT_TIMEOUT_MILLISECS = 180000L;
+
+  /**
+   * Get a thread pool based on a thread pool related property
+   *
+   * @param conf
+   *          accumulo configuration
+   * @param p
+   *          thread pool related property
+   * @return ExecutorService impl
+   * @throws RuntimeException
+   *           if property is not handled
+   */
+  public static ExecutorService getExecutorService(AccumuloConfiguration conf, Property p) {
```

Comment:
This method, and how you based it on the property might be my favorite piece of code this year. :smiley_cat:  This was a great way to bring all the various implementations into one centralized place.